In [62]:
import numpy as np
from scipy.stats import t
from scipy.stats import norm

# 1

In [68]:
Apple_Weights = [145, 155, 160, 146, 142, 152, 150, 147, 148, 149, 148, 152, 153, 155, 154, 148, 151, 147, 153, 146]

# hypothised mean weight of apple
as_mean = 150

#number of samples
n = len(Apple_Weights)
n

20

1a. What test should we use and why?

This is single sample.
We should use one-sample t-test because  one-sample t-test is used to test whether a population mean is equal to some hypothesized value.

In [69]:
apple_weights = np.array(Apple_Weights)

# sample mean and sample standard deviation
sample_mean = np.mean(apple_weights)
sample_std = np.std(apple_weights, ddof=1)

# assumed mean weight of apples
as_mean = 150

#  t-value
t_value = (sample_mean - as_mean) / (sample_std / np.sqrt(n))

#  p-value
p_value = 2 * t.sf(np.abs(t_value), n - 1)

print(f"Sample Mean : {sample_mean}")
print(f"Sample Standard deviation : {sample_std: .3f}")
print(f"t-value: {t_value:.3f}")
print(f"p-value: {p_value:.3f}")


Sample Mean : 150.05
Sample Standard deviation :  4.261
t-value: 0.052
p-value: 0.959


# 2

Suppose we want to test whether the mean height of all men in a population is 180 cm assuming that the population standard deviation = 2. We randomly sample 50 men from the population and measure their heights, getting the following data:

In [70]:
Men_height = [177, 180, 182, 179, 178, 181, 176, 183, 179, 180, 178, 181, 177, 178, 180, 179, 182, 180, 183, 181, 179, 177, 180, 181, 178, 180, 182, 179, 177, 182, 178, 181, 183, 179, 180, 181, 183, 178, 177, 181, 179, 182, 180, 181, 178, 180, 179, 181, 183, 179]

men_height = np.array(Men_height)

#assumed mean height of all men in population
as_mean1 = 180

#
assumed_std = 2

n1 = len(men_height)
n1

50

2a. What test should we use and why?

This is single sample of men height in a population, hence 
We should use one-sample t-test because  one-sample t-test is used to test whether a population mean is equal to some hypothesized value.

2b. State the null and alternative hypotheses.

H0 : mean height of all men equals to 180 cm

H1 : mean height of all men is not equal to 180cm

2c. Choose a significance level (α) (the probability of rejecting the null hypothesis when it is actually true).

We choose a significance level (α) of 0.05, which means we will accept 5 % chance or error


2d. Determine the degrees of freedom (df) of the sample. Df = sample size -1 

df = n1-1
df = 49

2e. Determine the critical value of t based on the significance level and degrees of freedom. 
For a two-tailed test with
α =  0.05 and df = 49,

the critical values of t is 2.009 and - 2.009

In [72]:
men_height = np.array(Men_height)
 
as_mean1 = 180

assumed_std = 2

n1 = len(men_height)


# sample mean and sample standard deviation
sample_mean1 = np.mean(men_height)
sample_std1 = np.std(men_height, ddof=1)

#  t-value
#t = (x̄ - μ) / (std / sqrt(n))

t_value1 = (sample_mean1 - as_mean1) / (sample_std1 / np.sqrt(n1))

#  p-value
p_value1 = 2 * t.sf(np.abs(t_value1), n1 - 1)

print(f"Sample Mean : {sample_mean1}")
print(f"Sample Standard deviation : {sample_std1: .3f}")
print(f"t-value: {t_value1:.3f}")
print(f"p-value: {p_value1:.3f}")


Sample Mean : 179.84
Sample Standard deviation :  1.867
t-value: -0.606
p-value: 0.547


# 3

In [27]:
Cats_weights = [3.9, 4.2, 4.5, 4.1, 4.3, 3.8, 4.6, 4.2, 3.7, 4.3, 3.9, 4.0, 4.1, 4.5, 4.2, 3.8, 3.9, 4.3, 4.1, 4.0, 4.4, 4.2, 4.1, 4.6, 4.4, 4.2, 4.1, 4.3, 4.0, 4.4, 4.3, 3.8, 4.1, 4.5, 4.2, 4.3, 4.0, 4.1, 4.2, 3.9, 4.3, 3.7, 4.1, 4.5, 4.2, 4.0, 4.2, 4.4, 4.1, 4.5]

cat_weights =np.array(Cats_weights)
n2 = len(Cats_weights)

#population mean weight of cats assumed
pop_mean = 4

#sample cat mean weight in kgs
sample_mean2 = np.mean(cat_weights)
print(f"Sample cat mean weight  = {sample_mean2:.3f}")

#sample cat std
sample_std2 = np.std(cat_weights, ddof = 1)
print(f"Sample cat std = {sample_std2:.3f}")


Sample cat mean weight  = 4.170
Sample cat std = 0.230


In [60]:
# calculating z score
# two-tailed Z-test with a significance level (α) of 0.05 (5%).

Z_score = (sample_mean2 - pop_mean)/(sample_std2/(np.sqrt(n2)))

print(f"Z score = {Z_score:.3f}")

Z score = 5.234


Since the calculated z-score of 5.23 does not lies between these critical z-values (ie - 1.96 and 1.96). We can  reject the null hypothesis. Therefore,  have sufficient evidence to conclude that the mean weight of the population of cats is significantly different from 4 kg at a significance level of 0.05.